tools/build_langserver/langserver/completion.go

```diff
@@ -26,25 +27,32 @@ func (h *LsHandler) handleCompletion(ctx context.Context, req *jsonrpc2.Request)
 	if err := json.Unmarshal(*req.Params, &params); err != nil {
 		return nil, err
 	}
-	log.Info("completion with param %s", req.Params)
+
 	documentURI, err := EnsureURL(params.TextDocument.URI, "file")
 	if err != nil {
+		message := fmt.Sprintf("invalid documentURI '%s' for method %s", documentURI, completionMethod)
+		log.Error(message)
 		return nil, &jsonrpc2.Error{
 			Code:    jsonrpc2.CodeInvalidParams,
-			Message: fmt.Sprintf("invalid documentURI '%s' for method %s", documentURI, completionMethod),
+			Message: fmt.Sprintf(message),
 		}
 	}
-	if !h.analyzer.IsBuildFile(documentURI) {
+	// Check if the input file is a build file or a build_defs file
+	if !h.analyzer.IsBuildFile(documentURI) || h.analyzer.IsBuildDefFile(documentURI) {
+		message := fmt.Sprintf("documentURI '%s' is not supported because it's not a buildfile", documentURI)
+		log.Error(message)
 		return nil, &jsonrpc2.Error{
 			Code:    jsonrpc2.CodeInvalidParams,
-			Message: fmt.Sprintf("documentURI '%s' is not supported because it's not a buildfile", documentURI),
+			Message: fmt.Sprintf(message),
 		}
 	}
 
-	supportSnippet := h.init.Capabilities.TextDocument.Completion.CompletionItem.SnippetSupport
-
 	h.mu.Lock()
-	itemList, err := getCompletionItemsList(ctx, h.analyzer, supportSnippet, documentURI, params.Position)
+	itemList, err := h.getCompletionItemsList(ctx, documentURI, params.Position)
+
+	if err != nil {
+		return nil, err
```

Comment:
WHOA

This is why you should always use `defer` for unlocking a mutex - if an error happens here you never unlock it and your process is probably borked forever.